src/main/java/de/retest/recheck/TestCaseFinder.java

```diff
@@ -86,7 +88,41 @@ public static TestCaseFinder getInstance() {
 	 * @return The class name for the test case method in all stack traces.
 	 */
 	public Optional<String> findTestCaseClassNameInStack() {
-		return findTestCaseMethodInStack( toClassName() );
+		final Optional<String> testCaseClassName = findTestCaseMethodInStack( toClassName() );
+		if ( testCaseClassName.isPresent() ) {
+			return testCaseClassName;
+		} else {
+			return findTestCaseClassInStack();
+		}
+	}
+
+	/**
+	 * @return The class name for the test being called from.
+	 */
+	public Optional<String> findTestCaseClassInStack() {
```

Comment:
Does this have to be public? If yes, I would add more documentation on what this method does, i.e. what `*Test` classes does it find. If no, I still would add some more doc to the above method.

Reply:
There is public `findTestCaseMethodInStack()`.